server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/TasksRoutesTest.java

```diff
@@ -101,22 +102,25 @@ void listShouldReturnTaskDetailsWhenTaskInProgress() throws Exception {
             .body("[0].status", is(TaskManager.Status.IN_PROGRESS.getValue()))
             .body("[0].taskId", is(taskId.asString()))
             .body("[0].class", is(not(empty())));
+
+        waitingForResultLatch.countDown();
```

Comment:
There's a problem with this solution: a failing test still leak resources.
What about handling the cleanup (here, calling countDown) in some @AfterEach method or writing an extension to manage the CountDownLatch?